File: nodejs/src/player/inputMod.js
const { Song } = require("./song")
const { stringIsValid, printFieldErrorMessage } = require("./utility")


module.exports.getSelection = async function (input) {
    console.log("==================================================")
    let selection

    try {
        selection = await input("Enter a selection (Q to quit): ")
        selection = selection.toUpperCase()

    } catch (e) {
        console.log(e)
        selection = ""
    } finally {
        return selection
    }
}

module.exports.getSong = async function (input) {
    let title, artist
    let rating = 0

    while (true) {
        title = await input("Enter song name: ")
        if (stringIsValid(title))
            break

    }



    while (true) {
        artist = await input("Enter song artist: ")
        if (stringIsValid(artist))
            break
    }

    while (true) {
        rating = await input("Enter your rating: ")
        if (stringIsValid(rating) && !isNaN(Number(rating)))
            break
        else
            console.log("Please enter a number")

    }

    const song = new Song(title, artist, rating)
    return song

}

File: src/lib/python/player/input_mod.py
from .song import Song
from .utility import string_is_valid, print_field_error_message


def get_selection():
    print("==================================================")
    selection = input("Enter a selection (Q to quit): ").upper()

    try:
        selection = selection.upper()
    except:
        selection = ""

    return selection


def get_song():
    title = ""
    artist = ""
    rating = 0

    while True:
        title = input("Enter song name: ")
        if string_is_valid(title):
            break

    while True:
        artist = input("Enter song artist: ")
        if string_is_valid(artist):
            break

    while True:
        try:
            rating = input("Enter your rating: ")
            int(rating)
            break
        except:
            print("Please enter a number")

    song = Song(title=title, artist=artist, rating=rating)
    return song
